framework/libra-framework/sources/ol_sources/lockbox.move

```diff
@@ -0,0 +1,464 @@
+/// rather than a fixed amount per day.
+/// It allows users to lock their assets for a specified period, ensuring that the assets
+/// cannot be accessed or transferred until the daily unlocks have drained the lockbox (the lock duration has expired).
+///
+/// ## Key Concepts
+///
+/// - **Time-Locked Assets**: Assets that are locked for a specific duration and cannot be accessed
+///   until the lock period ends.
+/// - **Lock Periods**: Predefined durations for which assets can be locked. These periods are
+///   represented as a vector of time intervals (in months).
+/// - **Daily Drip**: A mechanism that gradually unlocks assets from the lockbox on a daily basis,
+///   allowing partial access to the locked assets over time.
+///
+/// ## Constants
+///
+/// - `DEFAULT_LOCKS`: A vector of predefined lock periods (in months) that users can choose from.
+///   These periods are: 6 months, 12 months, 24 months, 36 months, 48 months, 96 months,
+///   192 months, 240 months, 288 months, 336 months, and 384 months.
+///
+/// ## Functions
+///
+/// The `Lockbox` module provides functions to:
+/// - Lock assets for a specified period.
+/// - Check the status of locked assets.
+/// - Unlock assets once the lock period has expired.
+/// - Gradually unlock assets on a daily basis through the daily drip mechanism.
+///
+///
+/// This module ensures that assets are securely locked and cannot be accessed until the specified
+/// lock period has expired, providing a mechanism for time-based asset management with gradual
+/// unlocking through the daily drip mechanism.
+
+
+module ol_framework::lockbox {
+  use std::error;
+  use std::fixed_point32::{Self, FixedPoint32};
+  use std::option::{Self, Option};
+  use std::signer;
+  use std::vector;
+  use diem_framework::coin::Coin;
+  use diem_std::math64;
+  use ol_framework::libra_coin::{Self, LibraCoin};
+  use ol_framework::date;
+
+  // use diem_framework::debug::print;
+
+  friend ol_framework::ol_account;
+
+  /// SlowWalletV2 not initialized
+  const ENOT_INITIALIZED: u64 = 1;
+
+  /// No lockbox of this duration found
+  const ENO_DURATION_FOUND: u64 = 2;
+
+  /// List of durations incomplete
+  const ELIST_INCOMPLETE: u64 = 3;
+
+  /// Tried to drip twice in one day, silly rabbit
+  const ENO_DOUBLE_DIPPING: u64 = 4;
+
+  /// Destination does not have SlowWalletv2 initialized
+  const EDESTINATION_NOT_INIT: u64 = 5;
+
+  /// Can only shift to a later duration to drip slower
+  const EMUST_SHIFT_LATER: u64 = 6;
+
+  /// Total balances should never change when shifting coins to later duration.
+  const EBALANCES_CHANGED_ON_DELAY: u64 = 7;
+
+  /// The locked units for this box duration should have changed.
+  const EDURATION_UNITS_SHOULD_CHANGE: u64 = 8;
+
+  /// Duration not in allowed list
+  const EINVALID_DURATION: u64 = 9;
+
+  const DEFAULT_LOCKS: vector<u64> = vector[6, 1*12, 2*12, 3*12, 4*12, 8*12, 16*12, 20*12, 24*12, 28*12, 32*12];
```

Comment:
Perhaps unnecessarily pedantic, but...these aren't "default" lock durations -- they appear to be "valid" or "mandatory" lock durations. You can't create a locked account with a duration that's not in this vector. Default would imply that if you don't provide a duration, you get one of the values in this vector. I think this difference between functionality and naming crosses the line into just the wrong variable name to use, unless I'm not reading the code correctly.